typedoc.json

```diff
@@ -0,0 +1,69 @@
+{
+	"$schema": "https://typedoc.org/schema.json",
+	"categoryOrder": [
+		"Node Types",
+		"Nodes - Type Guards",
+		"Nodes - Flag Utilities",
+		"Nodes - Other Utilities",
+		"Modifier Utilities",
+		"Scope Utilities",
+		"Syntax Utilities",
+		"Compiler Options",
+		"*",
+		"Other"
+	],
+	"customCss": "./typedoc.css",
+	"defaultCategory": "Other",
+	"entryPoints": ["./src/index.ts"],
+	"exclude": ["./**/test/", "./**/*.test.*"],
+	"excludeExternals": true,
+	"excludeInternal": true,
+	"excludeNotDocumented": false,
+	"excludePrivate": true,
+	"excludeProtected": true,
+	"hideGenerator": true,
+	"includeVersion": true,
+	"kindSortOrder": [
+		"Reference",
+		"Project",
+		"Module",
+		"Namespace",
+		"Constructor",
+		"Property",
+		"Variable",
+		"Function",
+		"Accessor",
+		"Method",
+		"ObjectLiteral",
+		"Parameter",
+		"Enum",
+		"EnumMember",
+		"Class",
+		"Interface",
+		"TypeAlias",
+		"TypeParameter",
+		"TypeLiteral",
+		"CallSignature",
+		"ConstructorSignature",
+		"IndexSignature",
+		"GetSignature",
+		"SetSignature"
+	],
+	"out": "./gh-pages",
```

Comment:
[Question] Hmm, I see `docs/` as a more standard folder locally. Most projects I've seen set up something like this use `docs/`. `gh-pages` is usually the branch name. Is there a reason to prefer this?

Reply:
I made this change simply so that we can write our own source docs files in the `docs` folder.

We're also using the newer way of using GitHub Pages that doesn't require us to push any built doc files to the repo. So ultimately the folder we build the docs in to doesn't matter (so long as nothing else is using it - aka it's empty) - the folder is never committed.

Reply:
Gotcha. I'd vote for `docs/generated/` as the place to put them, but not a blocker from me.